### PR TITLE
[DO NOT MERGE] Add support for all scalar coefficient types with libCEED.

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR adds support for `QuadratureFunctionCoefficient` and generic `Coefficient` evaluation on host when using libCEED.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1716](https://github.com/mfem/mfem/pull/1716) | @YohannDudouit | @v-dobrev | @psocratis + @artv3 | 08/26/20 | ⌛due 09/09/20 | ⌛due 10/07/20 | |

### CODE REVIEW

**Comment by artv3** [comment]
Is there a test for this? Or is there a recommended way to test the code?

**Comment by YohannDudouit** [comment]
The tests are being developed in #1719 , It will contain tests for all the libCEED integration features, that's why I separated them, but we could consider putting them back here.
Otherwise, I tested with `ex1` declaring a `double func_coeff(const Vector &x)` and using it in a `FunctionCoefficient`.

File: fem/libceed/ceed.cpp [hunk 1]
```diff
@@ -364,18 +455,32 @@ void CeedPAAssemble(const CeedPAOperator& op,
...
+         CeedQuadCoeff* quadCoeff = (CeedQuadCoeff*)ceedData.coeff;
+         const int ncomp = 1;
+         CeedInt strides[3] = {1, nqpts, ncomp*nqpts};
+         CeedElemRestrictionCreateStrided(ceed, nelem, nqpts, ncomp,
+                                          nelem*ncomp*nqpts, strides,
+                                          &quadCoeff->restr);
+         CeedOperatorSetField(ceedData.build_oper, "coeff", quadCoeff->restr,
+                              CEED_BASIS_COLLOCATED, quadCoeff->coeffVector);
+      }
+      break;
```
Comment (artv3) [inline fem/libceed/ceed.cpp:483]: Should there be default case?

Comment (YohannDudouit) [reply fem/libceed/ceed.cpp:483]: All cases are treated, the compiler would issue a warning otherwise.

Comment (YohannDudouit) [reply fem/libceed/ceed.cpp:483]: I should remove the `error` in the other `switch`

**Comment by YohannDudouit** [comment]
This branch is merged and tested in #1719.

**Comment by v-dobrev** [comment]
> This branch is merged and tested in #1719.

Does this mean you want close this PR and review the additions as part of #1719?

**Comment by YohannDudouit** [comment]
@v-dobrev I don't want to close this PR, I'm just merging all the libCEED PR into one because the conflicts are not simple to resolve. I think this way a reviewing is more efficient and more pleasant for the reviewers, and easier for merging for the editors.

**Comment by YohannDudouit** [comment]
@psocratis @artv3 Once you've reviewed and approved this PR, could you please check that the merge in #1719 looks good to you, and if it does write "The merge of #1716 in #1719 looks good to me."
I will close this PR afterwards in favor of #1719 to avoid complex merge conflicts for the editors.

**Comment by psocratis** [comment]
> @psocratis @artv3 Once you've reviewed and approved this PR, could you please check that the merge in #1719 looks good to you, and if it does write "The merge of #1716 in #1719 looks good to me."
> I will close this PR afterwards in favor of #1719 to avoid complex merge conflicts for the editors.

Sure, will do!

**Review by psocratis — APPROVED** [review]
Code looks good. Also, all ceed tests from #1719  are passing. 
The merge of #1716 in #1719 looks good to me.

**Comment by artv3** [comment]
The merge of #1716 in #1719 looks good to me.

**Comment by tzanio** [comment]
Can we delete the branch?

**Pull request closed without merging**